app/controllers/subscriptions_controller.rb

```diff
@@ -11,7 +11,10 @@ def new
 
   def create
     @subscription = current_user.subscriptions.build(subscription_params)
-    if @subscription.save
+    if @subscription.over_limit?
```

Comment:
The question over_limit? should be on the user level, not on subscription itself